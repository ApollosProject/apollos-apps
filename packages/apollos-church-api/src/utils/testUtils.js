import getContext from 'apollos-church-api/src/getContext';
import getDataSources from 'apollos-church-api/src/getDataSources';
import { KeyValueCache } from 'apollo-server-caching';

export function getTestContext(req) {
  const context = getContext(req);

  const dataSources = getDataSources();
  // Apollo Server does this internally.
  Object.values(dataSources).forEach((dataSource) => {
    if (dataSource.initialize) {
      dataSource.initialize({ context, cache: KeyValueCache });
    }
  });
  context.dataSources = dataSources;
  return context;
}

export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
