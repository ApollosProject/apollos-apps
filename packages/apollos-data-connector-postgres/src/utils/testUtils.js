/* eslint-disable import/prefer-default-export */
import createMigrationRunner from '../postgres/performMigrations';
import { sequelize } from '../postgres';

export const setupPostgresTestEnv = async (models = [], context) => {
  if (context.church.slug !== 'global') {
    await setupPostgresTestEnv(models, { church: { slug: 'global' } });
  }

  try {
    // If we don't have a set of migrations.
    // Hacky way of saying - have the migrations been run?
    await sequelize
      .getQueryInterface()
      .sequelize.query('select * from "SequelizeMeta"');
  } catch (e) {
    const migrationRunner = await createMigrationRunner({
      logger: null,
    });
    await migrationRunner.up();
  }

  await Promise.all(
    models.map((m) => m.models.createModel && m.models.createModel(context))
  );

  await Promise.all(
    models.map((m) => m.models.setupModel && m.models.setupModel(context))
  );
};
