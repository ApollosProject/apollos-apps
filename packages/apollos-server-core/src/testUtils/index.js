/* eslint-disable import/prefer-default-export */
import { makeExecutableSchema } from 'apollo-server';
import { testSchema } from '@apollosproject/data-schema';
import { KeyValueCache } from 'apollo-server-caching';
import { createApolloServerConfig } from '..';

export const createTestHelpers = (models) => {
  const serverConfig = createApolloServerConfig(models);
  const getContext = (req) => {
    const testContext = serverConfig.context(req);
    const testDataSources = serverConfig.dataSources();
    // Apollo Server does this internally.
    Object.values(testDataSources).forEach((dataSource) => {
      if (dataSource.initialize) {
        dataSource.initialize({ context: testContext, cache: KeyValueCache });
      }
    });
    testContext.dataSources = testDataSources;
    return testContext;
  };
  const getSchema = (schemas = []) =>
    makeExecutableSchema({
      typeDefs: [...serverConfig.schema, ...schemas, testSchema],
      resolvers: serverConfig.resolvers,
      resolverValidationOptions: {
        requireResolversForResolveType: false,
      },
    });

  return { getSchema, getContext };
};
