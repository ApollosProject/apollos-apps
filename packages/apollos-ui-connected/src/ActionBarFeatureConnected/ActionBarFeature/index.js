import React from 'react';
import {
  ActionBar,
  ActionBarItem,
  H4,
  styled,
  PaddedView,
  withTheme,
} from '@apollosproject/ui-kit';
import PropTypes from 'prop-types';

const ActionBarHeader = styled(
  {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  'ui-connected.ActionBarFeatureConnected.ActionBarHeader'
)(PaddedView);

const ActionBarFeature = ({ actions, title, onPressItem }) => (
  <>
    {title && (
      <ActionBarHeader vertical={false}>
        <H4>{title}</H4>
      </ActionBarHeader>
    )}
    <ActionBar>
      {actions.map((item) => (
        <ActionBarItem
          key={item.id}
          onPress={() => onPressItem(item)}
          icon={item.icon}
          label={item.title}
        />
      ))}
    </ActionBar>
  </>
);

ActionBarFeature.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  title: PropTypes,
  onPressItem: PropTypes.func,
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      label: PropTypes.string,
      icon: PropTypes.string,
    })
  ),
};

export default withTheme(
  () => ({}),
  'ui-connected.ActionBarFeatureConnected.ActionBarFeature'
)(ActionBarFeature);
