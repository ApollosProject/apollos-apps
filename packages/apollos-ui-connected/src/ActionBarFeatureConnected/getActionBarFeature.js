import gql from 'graphql-tag';
import ApollosConfig from '@apollosproject/config';

export default gql`
  query getActionBarFeature($featureId: ID!) {
    node(id: $featureId) {
      ...ActionBarFeatureFragment
    }
  }
  ${ApollosConfig.FRAGMENTS.ACTION_BAR_FEATURE_FRAGMENT}
  # Following fragment nested inside the ActionlistFeature
  ${ApollosConfig.FRAGMENTS.RELATED_NODE_FRAGMENT}
`;
