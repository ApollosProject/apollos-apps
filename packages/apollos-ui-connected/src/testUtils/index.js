/* eslint-disable import/prefer-default-export */
import { createStackNavigator, createAppContainer } from 'react-navigation';
import { IntrospectionFragmentMatcher } from 'apollo-cache-inmemory';
import apolloStorybookDecorator from 'apollo-storybook-react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from 'react-apollo/test-utils';
import possibleTypesJson from './fragmentTypes.json';
import typeDefs from './typeDefsMock';

const renderWithApolloData = async (component, existingTree) => {
  const tree = existingTree || renderer.create(component);
  await wait(1);
  tree.update(component);
  return tree;
};

// eslint-disable-next-line
const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);

const ApolloStorybookDecorator = ({
  mocks,
  additionalSchema = [],
  possibleTypes = {},
}) => {
  // We can use this way simpler code long term when we upgrade to Apollo 3.
  // const finalPossibleTypes = possibleTypes;
  // possibleTypesJson.__schema.types.forEach((supertype) => {
  //   if (supertype.possibleTypes) {
  //     finalPossibleTypes[supertype.name] = [
  //       ...supertype.possibleTypes.map((subtype) => subtype.name),
  //       ...(possibleTypes[supertype.name] || []),
  //     ];
  //   }
  // });
  const finalPossibleTypes = possibleTypesJson;
  Object.keys(possibleTypes).forEach((key) => {
    const jsonTypeIndex = possibleTypesJson.__schema.types.findIndex(
      ({ name }) => name === key
    );
    const newTypePossibleTypes = {
      ...possibleTypesJson.__schema.types[jsonTypeIndex],
      possibleTypes: [
        ...possibleTypesJson.__schema.types[jsonTypeIndex].possibleTypes,
        ...possibleTypes[key].map((type) => ({ name: type })),
      ],
    };
    possibleTypesJson.__schema.types[jsonTypeIndex] = newTypePossibleTypes;
  });
  const fragmentMatcher = new IntrospectionFragmentMatcher({
    introspectionQueryResultData: finalPossibleTypes,
  });

  return apolloStorybookDecorator({
    typeDefs: [typeDefs, ...additionalSchema],
    mocks,
    resolverValidationOptions: {
      requireResolversForResolveType: false,
    },
    cacheOptions: { fragmentMatcher },
  });
};

const WithReactNavigator = (Component) => {
  const AppNavigator = createStackNavigator({
    Home: {
      screen: () => Component,
      navigationOptions: {
        headerMode: 'none',
        header: null,
      },
    },
  });

  const AppContainer = createAppContainer(AppNavigator);
  return <AppContainer />;
};
export {
  renderWithApolloData,
  Providers,
  ApolloStorybookDecorator,
  WithReactNavigator,
};
