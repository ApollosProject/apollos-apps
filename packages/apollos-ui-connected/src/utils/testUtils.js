/* eslint-disable import/prefer-default-export */
import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from 'react-apollo/test-utils';

const renderWithApolloData = async (component, existingTree) => {
  const tree = existingTree || renderer.create(component);
  await wait(1);
  tree.update(component);
  return tree;
};

// eslint-disable-next-line
const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);

export { renderWithApolloData, Providers };
