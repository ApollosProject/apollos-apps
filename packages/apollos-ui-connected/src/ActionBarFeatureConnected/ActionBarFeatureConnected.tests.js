import React from 'react';
import renderer from 'react-test-renderer';

import { Providers, renderWithApolloData } from '../testUtils';

import GET_ACTION_BAR_FEATURE from './getActionBarFeature';
import ActionBarFeatureConnected from './index';

const mock = {
  request: {
    query: GET_ACTION_BAR_FEATURE,
    variables: { featureId: 'ActionBarFeature:123' },
  },
  result: {
    data: {
      node: {
        id: 'ActionBarFeatureConnected:123',
        __typename: 'ActionBarFeatureConnected',
        title: 'Some cool list',
        actions: [
          {
            __typename: 'ActionBarAction',
            id: 'ActionBarAction:123',
            title: 'Boom',
            icon: 'check',
            action: 'OPEN_URL',
            relatedNode: {
              __typename: 'Url',
              id: 'Url:123',
              url: 'http://www.google.com',
            },
          },
        ],
      },
    },
  },
};

describe('The ActionBarFeatureConnected component', () => {
  it('should render', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={[mock]}>
        <ActionBarFeatureConnected
          refetchRef={jest.fn()}
          featureId={'ActionBarFeature:123'}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state when isLoading', async () => {
    const tree = renderer.create(
      <Providers>
        <ActionBarFeatureConnected
          refetchRef={jest.fn()}
          featureId={'ActionBarFeature:123'}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
