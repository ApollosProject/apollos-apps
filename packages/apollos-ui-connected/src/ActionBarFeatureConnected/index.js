import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import ActionBarFeature from './ActionBarFeature';
import GET_ACTION_BAR_FEATURE from './getActionBarFeature';

const ActionBarConnected = ({ featureId, refetchRef, ...props }) => (
  <Query
    query={GET_ACTION_BAR_FEATURE}
    variables={{ featureId }}
    fetchPolicy="cache-and-network"
  >
    {({ data, refetch }) => {
      if (featureId && refetch && refetchRef)
        refetchRef({ refetch, id: featureId });
      const node = data?.node || {};
      return (
        <ActionBarFeature {...props} {...node} actions={node.actions || []} />
      );
    }}
  </Query>
);

ActionBarConnected.propTypes = {
  refetchRef: PropTypes.func,
  featureId: PropTypes.string,
};

export default ActionBarConnected;
