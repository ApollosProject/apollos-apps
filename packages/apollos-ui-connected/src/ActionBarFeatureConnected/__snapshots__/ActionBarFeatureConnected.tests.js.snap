// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ActionBarFeatureConnected component should render 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 0,
      }
    }
    vertical={false}
  >
    <Text
      style={
        Object {
          "color": "#303030",
          "fontFamily": "InterUI-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        }
      }
    >
      Some cool list
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 16,
        "marginHorizontal": 16,
        "marginVertical": 12,
        "shadowColor": "rgba(0, 0, 0, 0.09999999999999998)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 16,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "padding": 16,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              height={24}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  undefined,
                  null,
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width={24}
            >
              <RNSVGGroup
                fill={
                  Array [
                    0,
                    4278190080,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGPath
                  d="M19.18 6.32l-9.9 9.65-4.46-4.3c-.22-.23-.48-.33-.76-.33-.3 0-.54.1-.76.32-.2.2-.3.43-.3.7 0 .3.1.54.3.73l5.25 5.03c.2.22.44.32.74.32s.53-.1.72-.32l.14-.13L20.7 7.77c.2-.2.3-.43.3-.72 0-.3-.1-.53-.3-.72-.22-.2-.47-.32-.76-.32-.28 0-.54.1-.76.33z"
                  fill={
                    Array [
                      0,
                      4278216557,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
            <Text
              style={
                Object {
                  "color": "#00676D",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontStyle": null,
                  "fontWeight": null,
                  "lineHeight": 21,
                }
              }
            >
              Boom
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`The ActionBarFeatureConnected component should render a loading state when isLoading 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 16,
      "marginHorizontal": 16,
      "marginVertical": 12,
      "shadowColor": "rgba(0, 0, 0, 0.09999999999999998)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 16,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    />
  </View>
</View>
`;
